pandas/core/index.py

```diff
@@ -3945,20 +4026,15 @@ def get_indexer(self, target, method=None, limit=None):
 
         self_index = self._tuple_index
 
-        if method == 'pad':
-            if not self.is_unique or not self.is_monotonic:
-                raise AssertionError(('Must be unique and monotonic to '
-                                      'use forward fill getting the indexer'))
-            indexer = self_index._engine.get_pad_indexer(target_index.values,
-                                                         limit=limit)
-        elif method == 'backfill':
-            if not self.is_unique or not self.is_monotonic:
-                raise AssertionError(('Must be unique and monotonic to '
-                                      'use backward fill getting the indexer'))
-            indexer = self_index._engine.get_backfill_indexer(target_index.values,
-                                                              limit=limit)
+        if method == 'pad' or method == 'backfill':
+            indexer = self_index._get_fill_indexer(target, method, limit)
+        elif method == 'nearest':
+            raise NotImplementedError("method='nearest' not implemented yet "
+                                      'for MultiIindex; see GitHub issue 9365')
+        elif method is None:
+            indexer = self_index._engine.get_indexer(target.values)
         else:
-            indexer = self_index._engine.get_indexer(target_index.values)
+            raise ValueError('unrecognized method: %s' % method)
```

Comment:
maybe make a general check for the method (IIRC there was one somewhere), and do this first (and raise the ValueError), so the internal methods can be assured of a good method.